engine/flink/components/base-tests/src/test/scala/pl/touk/nussknacker/engine/flink/table/join/TableJoinTest.scala

```diff
@@ -183,16 +182,96 @@ class TableJoinTest
     }
   }
 
+  test("should allow to use various types in main branch context") {
+    val mainBranchLocalVariableFields = Map(
+      "nestedRecord"  -> "{foo: 1, bar: '123'}".spel,
+      "listOfRecords" -> "{{foo: 1, bar: '123'}}".spel,
+      "instant"       -> SpelValues.spelInstant,
+      // TODO: add support for types not supported in table api
```

Comment:
should it stay?

Reply:
for now yes, I'm trying to resolve this in the PR that I'm working on now